staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1365,9 +1367,36 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 		)
 	}
 
+	if opts.CountMetricPollPeriod > 0 {
+		stopFunc := e.startObservingCount(opts.CountMetricPollPeriod)
+		previousDestroy := e.DestroyFunc
+		e.DestroyFunc = func() {
+			stopFunc()
+			if previousDestroy != nil {
+				previousDestroy()
+			}
+		}
+	}
+
 	return nil
 }
 
+// startObservingCount starts monitoring given prefix and periodically updating metrics. It return a function to stop collection.
+func (e *Store) startObservingCount(period time.Duration) func() {
+	prefix := e.KeyRootFunc(genericapirequest.NewContext())
+	resourceName := e.DefaultQualifiedResource.String()
+	glog.V(2).Infof("Monitoring %v bound to <storage-prefix>/%v", resourceName, prefix)
+	stopCh := make(chan struct{})
+	go wait.Until(func() {
+		count, err := e.Storage.Count(prefix)
+		if err != nil {
+			glog.V(10).Infof("Failed to update storage count metric: %v", err)
```

Comment:
Why can this happen? 10 is pretty high for an error.

Reply:
It may happen if someone is using etcd2, which doesn't have sane implementation of Count. We can treat it more seriously when we drop etcd2 support:)

Reply:
don't we have a NotImplemented error type?